src/Type/JWT/DefaultJWTUserFactory.php

```diff
@@ -2,8 +2,10 @@
 
 namespace FlexAuth\Type\JWT;
 
-use Symfony\Component\Security\Core\User\User;
 use Symfony\Component\Security\Core\User\UserInterface;
+use UserBase\Client\Model\Account;
+use UserBase\Client\Model\AccountUser;
+use UserBase\Client\Model\User;
```

Comment:
Suggest you create JWTUserbaseFactory class instead of change default one

Reply:
Then we need to modify flex-auth-provider & flex-auth-bundle. We're changing implementation, signature stays the same, so that's fine. Is there any reasons for leaving this particular implementation? (if so we should also add configuration for both repos). And in fact we need to have userbase structure because all current consumers of this library depend on it
Do you know cases for using plain Symfony user implementation?

Reply:
Flex Auth is based on symfony security only.
JWT type includes common implementation of JWT and standalone from any libs expect symfony security. Actually UserBase lib is optional dependency.
It is why I use core user class by default and it make more friendly for other developers.
Primary feature is change security providers dynamically without any integration by default.

I did't see any problem to add custom `JWTUserbaseFactory` which implemented `JWTUserFactoryInterface`. And specify that explicit for your project. 

In particulary `flex-auth-bundle` use autowire for `JWTTokenAuthenticator`.
And you can just add to `services.yml`
```yml
FlexAuth\Type\JWT\JWTUserFactoryInterface: 'FlexAuth\Type\JWT\JWTUserBaseFactory'
```

Reply:
Primary goal of this library as far as I know was to replace built-in userbase auth with a different options. So that's not really 'optional' dependency. But it's a public repo now. So ideally we need to implement some kind of configuration for Symfony & Silex integration. Or we could just achieve primary goal and replace main implementation for factory and then implement configuration for it.
If you want to make it developer friendly then just create `SymfonyJWTFactory` and leave opportunity to use this implementation for them by adding configuration to `flex-provider` and `flex-auth-bundle`.
But primary factory for all apps should be `UserBaseJWTFactory`. So this changes should stay and Symfony-based user factory with following configurations for bundle & silex provider should be implemented as additional improvements.

Reply:
I did't see any mention of `primary goal is replace built-in userbase auth` and `But primary factory for all apps should be UserBaseJWTFactory` in task description from @joostfaassen. I asked him but hi did't reply me yet.
We don't need to create `SymfonyJWTFactory`. `JWTTokenAuthenticator` implement base functional already. If you need customize something you can replace some injected services.
In particularly I didn't see any problem to implement `JWTUserBaseFactory` and specify in `services.yml` explicitly as I mean above.

Reply:
Hi @sashaaro and @rost1994, interesting discussion, and thanks for the good input. I can see there are good arguments for both routes.

As rost is saying: yes, we will use this library in applications that currently depend on userbase. But, that is not a good dependency, I think those apps should ideally be updated so they can work using "vanilla" symfony users. In most of our apps we don't even use the userbase specific methods (for accounts etc) so that should not be too complicated.

I'd like to think of the flex-auth libraries as generic and usable by many other (symfony) developers / projects.

So I agree that the current DefaultJWTUserFactory can stay as it is, and remains the default. A new UserBaseJWTFactory can be used in applications that are strongly tied to userbase, until we've refactored those to be able to work without it.

Hope this helps!

Reply:
Sorry about the delay btw.. i've been away for a few days, but back to normal, so if there are any updates I'll be around for quicker answers.. Thanks!

Reply:
Hi @joostfaassen . Sure